eng/Directory.Build.Data.targets

```diff
@@ -120,7 +120,7 @@
 
   <ItemGroup Condition="'$(UseProjectReferenceToAzureClients)' == 'true'">
     <!-- Find all Azure projects -->
-    <AzureProjects Include="$(RepoRoot)/sdk/*/Azure.*/src/Azure.*.csproj" ProjectPath="%(Identity)" />
+    <AzureProjects Include="$(RepoRoot)/sdk/**/src/*.csproj" ProjectPath="%(Identity)" />
```

Comment:
cc @weshaggard 

We have track 2 projects that don't start with Azure.

Reply:
OK I think this should be OK to change it should in theory work for any package reference we have a project for in the repo.